scanner/src/main/kotlin/scanners/scancode/ScanCode.kt

```diff
@@ -140,9 +142,12 @@ class ScanCode internal constructor(
     override fun transformVersion(output: String): String {
         // On first use, the output is prefixed by "Configuring ScanCode for first use...". The version string can be
         // something like:
-        // ScanCode version 2.0.1.post1.fb67a181
-        val prefix = "ScanCode version "
-        return output.lineSequence().first { it.startsWith(prefix) }.substring(prefix.length)
+        // ScanCode version 2.0.1.post1.fb67a181 or ScanCode version: 31.0.0b4
+        return output.lineSequence().first { it.matches(versionRegex) }.let {
+            val prefix = versionRegex.find(it)!!.groups[1]!!.value
+
+            it.removePrefix(prefix)
```

Comment:
Just throwing in an alternative implementation (to get rid of the ugly `!!` casts):

        return output.lineSequence().firstNotNullOfOrNull { line ->
            line.withoutPrefix("ScanCode version")?.removePrefix(":")?.trim()
        }.orEmpty()

Reply:
So here's another idea to get rid of `!!`:

        private val versionRegex = """^ScanCode version:? (?<version>.+)""".toRegex()

        return output.lineSequence().firstNotNullOfOrNull {
            versionRegex.matchEntire(it)?.groups?.get("version")?.value
        }.orEmpty()